### PR TITLE
[FakeSlave] Extend slaves to provide period data for properly testing PDOs

### PR DESCRIPTION
**Pull request by destogl**
We have extended FakeSlave to publish one more PDO asynchronously from everything else to extend and test capability of ros2_control components.

### CODE REVIEW

File: canopen_fake_slaves/config/simple_slave.eds [hunk 1]
```diff
@@ -267,4 +267,4 @@ PDOMapping=1
 ParameterName=UNSIGNED32 sent from slave
 DataType=0x0007
 AccessType=rwr
-PDOMapping=1
+PDOMapping=1
```
Comment (destogl) [inline canopen_fake_slaves/config/simple_slave.eds:270]: Unnötige Änderung.

```suggestion
PDOMapping=1

```

File: canopen_fake_slaves/include/canopen_fake_slaves/basic_slave.hpp [hunk 2]
```diff
@@ -49,6 +57,21 @@ class SimpleSlave : public canopen::BasicSlave
     uint32_t val = (*this)[idx][subidx];
     (*this)[0x4001][0] = val;
     this->TpdoEvent(0);
+    // Publish periodic message
+    message_thread = std::thread(std::bind(&SimpleSlave::fake_periodic_messages, this));
+  }
+
+  void fake_periodic_messages()
```
Comment (destogl) [inline canopen_fake_slaves/include/canopen_fake_slaves/basic_slave.hpp:64]: Please add a short docstring for this method.

File: canopen_fake_slaves/include/canopen_fake_slaves/basic_slave.hpp [hunk 3]
```diff
@@ -49,6 +57,21 @@ class SimpleSlave : public canopen::BasicSlave
     uint32_t val = (*this)[idx][subidx];
     (*this)[0x4001][0] = val;
     this->TpdoEvent(0);
+    // Publish periodic message
+    message_thread = std::thread(std::bind(&SimpleSlave::fake_periodic_messages, this));
```
Comment (destogl) [inline canopen_fake_slaves/include/canopen_fake_slaves/basic_slave.hpp:61]: We should not create a new tread each time. For now, a small `if` should help. An idea would be:

```suggestion
    if not message_thread:
        message_thread = std::thread(std::bind(&SimpleSlave::fake_periodic_messages, this));
```

Also, what I am not sure. You are using here class member, but the method does not have self. We should check `global` or similar attribute for the variable should be used.

File: canopen_fake_slaves/include/canopen_fake_slaves/basic_slave.hpp [hunk 4]
```diff
@@ -36,11 +36,19 @@ class SimpleSlave : public canopen::BasicSlave
...
+  ~SimpleSlave()
+  {
+    if (message_thread.joinable())
+    {
+      message_thread.join();
+    }
+  }
 
 protected:
+  std::thread message_thread;
```
Comment (destogl) [inline canopen_fake_slaves/include/canopen_fake_slaves/basic_slave.hpp:48]:
```suggestion
  std::thread message_thread = nullptr;
```

`nullptr` is valid in C++, not sure about python. I think it was only `null`;

**Review by destogl — COMMENTED** [review]
Just a few small fixes needed. Can you also please update the documentation with the explanation for users what they can expect here.

File: canopen_fake_slaves/include/canopen_fake_slaves/basic_slave.hpp [hunk 5]
```diff
@@ -36,11 +36,19 @@ class SimpleSlave : public canopen::BasicSlave
...
+  ~SimpleSlave()
+  {
+    if (message_thread.joinable())
+    {
+      message_thread.join();
+    }
+  }
 
 protected:
+  std::thread message_thread = nullptr;
```
Comment (destogl) [inline canopen_fake_slaves/include/canopen_fake_slaves/basic_slave.hpp:48]: This does not work on humble. Maybe by putting this into `unique_ptr` could be a solution. Or even simpler, but not so nice, adding a new flag variable.

**Review by destogl — COMMENTED** [review]
This looks good but please fix compilation and formatting.

**Review by ipa-vsp — APPROVED** [review]

**Pull request merged**